src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -233,9 +233,12 @@ internal void RemoveProject(AbstractProject project)
 
             UpdateProjectBinPath(project, project.TryGetBinOutputPath(), null);
 
-            if (project.PushingChangesToWorkspaceHosts)
+            using (Dispatcher.CurrentDispatcher.DisableProcessing())
             {
-                NotifyWorkspaceHosts(host => host.OnProjectRemoved(project.Id));
+                foreach (var hostState in _workspaceHosts)
```

Comment:
No, because `_pushedProjects.Remove(project)` that I'm adding above will return false.